cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java

```diff
@@ -123,19 +123,15 @@ public void createCollection(String dbName, String collectionName, long dataExpi
                 default:
              }
             options = new IndexOptions().name("cyg_agg_opt");
-            db.getCollection(collectionName).createIndex(keys, options);
-        } catch (Exception e) {
-            LOGGER.warn("Error in collection " + collectionName + " creating index ex=" + e.getMessage());
-        } // try catch
-        try {
-            if (dataExpiration != 0) {
-                keys = new BasicDBObject().append("_id.origin", 1);
-                options = new IndexOptions().name("cyg_agg_exp").expireAfter(dataExpiration, TimeUnit.SECONDS);
-                db.getCollection(collectionName).createIndex(keys, options);
-            } // if
+            createIndex(db, collectionName, keys, options);
         } catch (Exception e) {
             LOGGER.warn("Error in collection " + collectionName + " creating index ex=" + e.getMessage());
```

Comment:
Should the try/catch be removed (given that createIndex() will deal with the exception internally)?

Reply:
Fixed in 1458daf

Reply:
It seems the try/catch block is still there... or maybe I'm missing something?

Reply:
Fixed in b6045a0